dialogue-clients/src/main/java/com/palantir/dialogue/clients/ChannelCache.java

```diff
@@ -75,8 +75,14 @@ static ChannelCache createEmptyCache() {
         LIVE_INSTANCES.add(newCache);
 
         int numLiveInstances = LIVE_INSTANCES.size();
-        if ((numLiveInstances > 1 && log.isInfoEnabled()) || log.isDebugEnabled()) {
+        if ((numLiveInstances >= 10 && log.isInfoEnabled())) {
             log.info(
+                    "Created ChannelCache instance #{} ({} alive): {}",
+                    SafeArg.of("instanceNumber", newCache.instanceNumber),
+                    SafeArg.of("totalAliveNow", numLiveInstances),
+                    SafeArg.of("newCache", newCache));
```

Comment:
I think we still need a stack trace in this path, otherwise it's hard to figure out where these things are allocated. Also, if we have >=10 instances, updating the log level from INFO to DEBUG won't have any impact.

I think if debug is enabled, we record a stack trace in all cases, regardless of the `numLiveInstances` (just like before).